### DISCUSSION TITLE
Minimize Latency

### DISCUSSION THREAD

**Discussion by gebhard73** [Security-Onion-Solutions/securityonion#3105] [Unsupported Versions] [answered]
Hi,

for a special use case we've a dedicated system which only gets syslog. No traffic monitoring takes place. 

We want to keep the latency from the syslog events hitting the system to having them displayed in Kibana as short as possible and reasonable. 

Which would be the parameters to tune the pipeline for this use case? Any hint is appreciated.

Many thanks!
Gebhard

**Comment by weslambert** [comment, marked as answer]
You can try adjusting the following:

- Logstash Redis output (manager) - `batch_events`
- Logstash Redis input (search) - `batch_count`, `threads`
- `logstash.yml` - `pipeline.batch.size`

Keep in mind, these values should be adjusted in the pillar, when able.

**Reply by gebhard73** [reply]
many thanks, appreciated

**Reply by NRDCS-support** [reply]
Also change index refresh interval for index, there syslog messages are written. 
Setting can be set per each minion, in minion pillar  file ( /opt/so/saltstack/local/pillar/minion/...
By default setting here is is empty, indexes are refreshed using default value '30' sec.
So, you can change reftresh interval in minion file, by adding setting.

Another possibility is to change default value, coded in SO. Not reccomended, since setting can be lost after update.

If interested - this is written at the top of each index template file and applied when index template file is parsed by configuration management (salt). ( in my example default value is 60 seconds.)
{%- set REFRESH = salt['pillar.get']('elasticsearch:index_settings:so-zeek:refresh', '60s') %}

Do not set to low value ( bellow 10, not sure, test by yourself ) , as index refresh is considered to be a costly operation.

**Reply by gebhard73** [reply]
many thanks, will also test this one

**Reply by weslambert** [reply]
As mentioned above, I would be careful with adjusting refresh interval if you are ingesting large amounts of data. YMMV.

**Reply by gebhard73** [reply]
Thanks, will do.
